src/instances.ts

```diff
@@ -0,0 +1,154 @@
+import axios from 'axios';
+import fs from 'fs';
+import path from 'path';
+
+
+interface TokenInstance {
+    id: string;
+    is_unique: boolean;
+    owner: {
+        hash: string;
+    };
+}
+
+interface ApiResponse {
+    items: TokenInstance[];
+    next_page_params: {
+        index: string;
+        items_count: string;
+    } | null;
+}
+
+interface ProjectConfig {
+    address: string;
+    tokenURIchangeNeeded: string;
+}
+
+interface TokenCounters {
+    token_holders_count: string;
+    transfers_count: string;
+}
+const BASE_API_URL = 'https://astar-zkevm.explorer.startale.com/api/v2';
+
+/**
+ * Fetches all token instances for a given contract address
+ * @param contractAddress The NFT contract address
+ * @returns Map of tokenId to owner address
```

Comment:
Will this work for Yoki (ERC 1155)?

Reply:
good point, but I left this for last bit of the project. I am expecting it will not work for 1155